### DISCUSSION TITLE
Heroku + PostgreSQL issue

### DISCUSSION THREAD

**Discussion by sylvaindeloux** [directus/directus#5047] [Archive]
Hi

I'm using Heroku, with a PostgreSQL add-on and I get "Can't connect to the database" error during release process (no issue during build).
All my env are in the `directus.config.js` file and I set the `CONFIG_PATH` to `/app/directus.config.js` on dyno settings.

My `package.json`:

```json
{
  "dependencies": {
    "directus": "^9.0.0-rc.53",
    "pg": "^8.6.0"
  }
}
```

My `directus.config.js`:

```
module.exports = {
    ...
    DB_CLIENT: "pg",
    ...
    DB_CONNECTION_STRING: process.env.DB_CONNECTION_STRING || process.env.DATABASE_URL,
    ...
};
```

Database DSN is added automatically in `DATABASE_URL` by Heroku.

Result (release log):

```
06:35:36 ✨ Initializing bootstrap...
06:36:01 🚨 Can't connect to the database
```

Does someone have an idea? Thanks :)

**Comment by rijkvanzanten** [comment]
The only thing I can think of immediately would be a mismatch in Heroku's string format, a wrong environment variable name, or maybe that config file being wrong? 🤔

**Comment by dickyj** [comment]
I am using Docker Directus, and for some reason, after upgrading to a new docker release, I get the exact error as shown above "Can't connect to the database". Then I tried downgrading to previous version, and yet encountered the same error. 
My database is working just fine. Is there any way to find out the error code associated with this. I cannot seem to find any verbose mode.

**Comment by casey-chow** [comment]
If you're using a production-tier Heroku Postgres instance, you're probably hitting an SSL verification issue. What I had to do to get my instance stood up was the following in my config:

```js
  DB_CONNECTION_STRING:
    (process.env.DATABASE_CONNECTION_POOL_URL || process.env.DATABASE_URL) +
    '?sslmode=no-verify',
```

See more information in the [pg-connection-string docs](https://github.com/brianc/node-postgres/tree/master/packages/pg-connection-string#tcp-connections).
